### PR TITLE
Rewrite/Enhance/Improve docs for 0.21.0 release

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: src/docs/03-concepts/08-cross-dc-replication.md [hunk 1]
```diff
@@ -87,6 +87,79 @@ The following command can be used to failover Global :domain:Domain: *my-domain-
...
+
+The best way is to use Cadence [docker-compose](https://github.com/uber/cadence/tree/master/docker):
+`docker-compose -f docker-compose-multiclusters.yml up`
+
+
+## Running in Production
+
+Enable global domain feature needs to be enabled in [static config](/docs/operation-guide/setup/#static-configs).  
+
+Here we use clusterDCA and clusterDCB as an example. We pick clusterDCA as the master cluster.
```
Comment (yux0) [inline src/docs/03-concepts/08-cross-dc-replication.md:100]: I will update this name soon. Could we use primary cluster?

**Review by just-at-uber — APPROVED** [review]
LGTM

File: src/docs/07-operation-guide/02-maintain.md [hunk 2]
```diff
@@ -69,3 +69,37 @@ Also, the schema tool by default will upgrade schema to the latest, so no manual
 
 Database schema changes are versioned in the folders: [Versioned Schema Changes](https://github.com/uber/cadence/tree/master/schema/mysql/v57/cadence/versioned) for Default Store
 and [Versioned Schema Changes](https://github.com/uber/cadence/tree/master/schema/mysql/v57/visibility/versioned) for Visibility Store if you use database for basic visibility instead of ElasticSearch.
+
+## Migrate Cadence cluster
+Migrating a Cadence cluster is rare, but could happen.
+There could be some reasons like:
+* Scale up -- move to a bigger cluster, with larger number of shards
```
Comment (yux0) [inline src/docs/07-operation-guide/02-maintain.md:76]: Is this true? I think our hashing will change if the number of shards changed. So we have to use the same number in two clusters.

File: src/docs/03-concepts/08-cross-dc-replication.md (hunk 1 above)
Comment (longquanzheng) [reply src/docs/03-concepts/08-cross-dc-replication.md:100]: yeah sure, will update it. 
btw when you update the configuration, we should have some release notes to tell people about it. and it's better to have a compatible period(minor) version with warnings logs before breaking.

File: src/docs/07-operation-guide/02-maintain.md (hunk 2 above)
Comment (longquanzheng) [reply src/docs/07-operation-guide/02-maintain.md]: Thanks, I filed an issue and add a TODO in this sentense.

**Pull request merged**